vaadin-number-field.html

```diff
@@ -0,0 +1,199 @@
+-->
+
+<link rel="import" href="../polymer/polymer-element.html">
+<link rel="import" href="../vaadin-themable-mixin/vaadin-themable-mixin.html">
+<link rel="import" href="vaadin-text-field.html">
+
+<dom-module id="vaadin-number-field-default-theme">
+</dom-module>
+
+<dom-module id="vaadin-number-field-template">
+  <template>
+    <style>
+      :host {
+        width: 120px;
+      }
+
+      :host([has-controls]) [part="value"] {
+        text-align: center;
+      }
+
+      /* Value control styles */
+      [part="decrease-button"],
+      [part="increase-button"] {
+        display: inline-block;
+        cursor: pointer;
+        padding: 0 8px;
+        opacity: 0.5;
+      }
+
+      [part="decrease-button"][hidden],
+      [part="increase-button"][hidden] {
+        display: none;
+      }
+
+      [part="decrease-button"]::before {
+        content: "−";
+      }
+
+      [part="increase-button"]::before {
+        content: "+";
+      }
+
+      /* Hide the native arrow icons */
+      [part="value"]::-webkit-outer-spin-button,
+      [part="value"]::-webkit-inner-spin-button {
+        -webkit-appearance: none;
+        margin: 0;
+      }
+
+      [part="value"] {
+        -moz-appearance: textfield;
+      }
+    </style>
+
+    <div
+      part="decrease-button"
+      on-click="_decreaseValue"
+      hidden$="[[!hasControls]]">
+    </div>
+
+    <div
+      part="increase-button"
+      on-click="_increaseValue"
+      hidden$="[[!hasControls]]">
+    </div>
+  </template>
+
+  <script>
+    (function() {
+      /**
+      * `<vaadin-number-field>` is a Polymer 2 element for number field control in forms.
+      *
+      * ```html
+      * <vaadin-number-field label="Number">
+      * </vaadin-number-field>
+      * ```
+      *
+      * @memberof Vaadin
+      * @extends Vaadin.TextFieldElement
+      * @demo demo/index.html
+      */
+
+      let memoizedTemplate;
+
+      class NumberFieldElement extends Vaadin.TextFieldElement {
+        static get is() {
+          return 'vaadin-number-field';
+        }
+
+        static get properties() {
+          return {
+            /**
+            * Set to true to display value increase/decrease controls.
+            */
+            hasControls: {
+              type: Boolean,
+              value: false,
+              reflectToAttribue: true
+            },
+
+            /**
+            * The minimum value of the field.
+            */
+            min: {
+              type: Number,
+              reflectToAttribue: true,
+              observer: '_minChanged'
+            },
+
+            /**
+             * The maximum value of the field.
+             */
+            max: {
+              type: Number,
+              reflectToAttribue: true,
+              observer: '_maxChanged'
+            }
+          };
+        }
+
+        ready() {
+          super.ready();
+          this.$.input.type = 'number';
```

Comment:
`this.focusElement.type = 'number';`

Reply:
Done, thanks for noticing this! I'll try to break through our bureaucracy and get `vaadin-number-field` merged even if it's not in our current roadmap.

Reply:
Your welcome! :)

Sounds good!

Reply:
@limonte offtopic question: Why Reviewable instead of GitHub reviews?

Reply:
Reviewable works better for our internal processes. For external contributions, we are trying to use GitHub comments.

Reply:
Ok, thanks!